Possible JSON circular reference which makes the result fail

### Discussed in https://github.com/lowlighter/metrics/discussions/607

<div type='discussions-op-text'>

<sup>Originally posted by **jayvdb** October 24, 2021</sup>
I added an `output_format: json` step to my CI, and it has failed three times
https://github.com/jayvdb/jayvdb/commit/0969aafdabf1ba3d94b3c7f58ed26eaa230c2a58
It is simpler than the SVG settings, so it is a bit surprising that it failed when the SVG was happily building.

When I remove 'indepth', I get a different error

> Error: Converting circular structure to JSON
>    --> starting at object with constructor 'Object'
>    |     property 'question' -> object with constructor 'Object'
>    --- property 'answer' closes the circle
> Error: Process completed with exit code 1.

Removing StackOverflow fixed that.</div>

--- a/source/app/metrics/index.mjs
+++ b/source/app/metrics/index.mjs
@@ -77,7 +77,20 @@ export default async function metrics({login, q}, {graphql, rest, plugins, conf,
     //JSON output
     if (convert === "json") {
       console.debug(`metrics/compute/${login} > json output`)
-      return {rendered:data, mime:"application/json"}
+      const cache = new WeakSet()
+      const rendered = JSON.parse(JSON.stringify(data, (key, value) => {
+        if ((value instanceof Set)||(Array.isArray(value)))
+          return [...value]
+        if (value instanceof Map)
+          return Object.fromEntries(value)
+        if ((typeof value === "object")&&(value)) {
+          if (cache.has(value))
+            return Object.fromEntries(Object.entries(value).map(([k, v]) => [k, cache.has(v) ? "[Circular]" : v]))
+          cache.add(value)
+        }
+        return value
+      }))
+      return {rendered, mime:"application/json"}
     }
 
     //Markdown output

